Shard test_nn to reduce runtime for each test target

yf225 opened the pull request:
Summary: The current test_nn would time out and be disabled in GreenWarden, and we need to have an option to split it up in order to pass the stress test. Right now GreenWarden roughly allows running 100 test cases in test_nn before timing out, and here we have an option to divide test_nn into 30 shards (with ~40 tests in each shard) to allow for some test suite growth in the future.

Differential Revision: D8592394

orionr commented:
Test failures are all caused by fbsync branch itself not passing right now. Going to land anyway.

Commit:
Shard test_nn to reduce runtime for each test target (#8793)

Summary:
Closes https://github.com/pytorch/pytorch/pull/8793

The current test_nn would time out and be disabled in GreenWarden, and we need to have an option to split it up in order to pass the stress test. Right now GreenWarden roughly allows running 100 test cases in test_nn before timing out, and here we have an option to divide test_nn into 30 shards (with ~40 tests in each shard) to allow for some test suite growth in the future.

Reviewed By: orionr

Differential Revision: D8592394

fbshipit-source-id: caa32f0a8d1b84c4b0a00006e8cba523e44de63b
